packages/edit-site/src/components/sidebar-navigation-screen-page/index.js

```diff
@@ -26,34 +113,158 @@ export default function SidebarNavigationScreenPage() {
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'attachment',
+						record?.featured_media
+				  )
+				: null;
+
+			// Parent page title.
+			const parent = record?.parent
+				? select( coreStore ).getEntityRecord(
+						'postType',
+						'page',
+						record.parent
+				  )
+				: null;
+			let _parentTitle = __( 'Top level' );
+			if ( parent ) {
+				_parentTitle = parent?.title?.rendered
+					? decodeEntities( parent.title.rendered )
+					: __( 'Untitled' );
+			}
+
+			return {
+				parentTitle: _parentTitle,
+				templateSlug: getEditedPostContext()?.templateSlug,
+				featuredMediaDetails: {
+					...getMediaDetails( attachedMedia, postId ),
+					mediaDescription: attachedMedia?.description?.raw,
+				},
+			};
+		},
+		[ record ]
+	);
+
+	// Match template slug to template title.
+	const { records: templates, isResolving: areTemplatesLoading } =
+		useEntityRecords( 'postType', 'wp_template', {
+			per_page: -1,
+		} );
+	const templateTitle =
+		! areTemplatesLoading && templateSlug
+			? templates?.find( ( template ) => template?.slug === templateSlug )
+					?.title?.rendered || templateSlug
+			: '';
+
+	const displayLink = record?.link
+		? record.link.replace( /^(https?:\/\/)?/, '' )
+		: null;
+
 	return (
 		<SidebarNavigationScreen
-			title={ record ? decodeEntities( record?.title?.rendered ) : null }
+			title={
+				record
+					? decodeEntities( record?.title?.rendered )
+					: __( 'Untitled' )
+			}
 			actions={
 				<SidebarButton
 					onClick={ () => setCanvasMode( 'edit' ) }
 					label={ __( 'Edit' ) }
 					icon={ pencil }
 				/>
 			}
-			description={ __(
-				'Pages are static and are not listed by date. Pages do not use tags or categories.'
-			) }
+			meta={
+				!! record?.link && (
+					<ExternalLink
+						className="edit-site-sidebar-navigation-screen__page-link"
+						href={ record.link }
+					>
+						{ displayLink }
+					</ExternalLink>
+				)
+			}
 			content={
 				<>
-					{ record?.link ? (
-						<ExternalLink
-							className="edit-site-sidebar-navigation-screen__page-link"
-							href={ record.link }
+					<VStack
+						className="edit-site-sidebar-navigation-screen-page__featured-image-wrapper"
+						alignment="left"
+						spacing={ 2 }
+					>
+						<div
+							className={ classnames(
+								'edit-site-sidebar-navigation-screen-page__featured-image',
+								{
+									'has-image': !! mediaSourceUrl,
+								}
+							) }
+						>
+							{ !! mediaSourceUrl && (
+								<img
+									alt={
+										mediaDescription
+											? decodeEntities( mediaDescription )
+											: decodeEntities(
+													record?.title?.rendered
+											  ) || __( 'Featured image' )
+									}
+									src={ mediaSourceUrl }
+								/>
+							) }
+							{ record && ! record?.featured_media && (
+								<p>{ __( 'No featured image' ) }</p>
+							) }
+						</div>
+					</VStack>
+					{ !! record?.excerpt?.raw && (
```

Comment:
Same comment regarding `raw` vs `rendered`.

Reply:
We'll have to use `stripHTML` instead because it comes with `<p>` tags. `dangerouslySetInnerHTML` doesn't appear to work with `<Truncate />`